api/scripts/certification/target-profile-to-calibrated-framework.js

```diff
@@ -0,0 +1,46 @@
+import { knex } from '../../db/knex-database-connection.js';
+import { usecases } from '../../src/certification/configuration/domain/usecases/index.js';
+import { ComplementaryCertificationKeys } from '../../src/certification/shared/domain/models/ComplementaryCertificationKeys.js';
+import { Script } from '../../src/shared/application/scripts/script.js';
+import { ScriptRunner } from '../../src/shared/application/scripts/script-runner.js';
+
+export class TargetProfileToCalibratedFrameworkScript extends Script {
+  constructor() {
+    super({
+      description: 'Use target profile to create a consolidated framework',
+      permanent: false,
+      options: {
+        complementaryCertificationKey: {
+          type: 'string',
+          describe: 'Certification key',
+          requiresArg: true,
+        },
+        targetProfileId: {
+          type: 'number',
+          describe: 'Target profile ID',
+          requiresArg: true,
+        },
+      },
+    });
+  }
+
+  async handle({ options, logger }) {
+    logger.info(`① Check ${options.complementaryCertificationKey} existence in the domain`);
+    if (!Object.values(ComplementaryCertificationKeys).includes(options.complementaryCertificationKey)) {
+      throw new Error('The certification key is missing');
+    }
+
+    logger.info(`② Retrieve target-profile #${options.targetProfileId} tubes`);
+    const tubeIds = await knex('target-profile_tubes')
+      .pluck('tubeId')
+      .where('targetProfileId', options.targetProfileId);
+
+    logger.info(`③ Handle retrieving tubes to create ${tubeIds.length} new certification frameworks challenges`);
```

Comment:
Ca me fait bizarre d'avoir une erreur si pas de ref cadre trouve, mais pas d'erreur si pas de tubes trouves.
Je propose d'empecher d'aller + loin si pas de tubes. Car le usecase empeche pas cela.

```suggestion
    if(tubeIds?.length < 1){
        throw new RangeError('This target profile does not hold any tubes');
    }

    logger.info(`③ Handle retrieving tubes to create ${tubeIds.length} new certification frameworks challenges`);
```